### PR TITLE
Improve CI and fix examples

### PR DESCRIPTION
**Pull request by SergioGasquez**
- CI is now run in parallel
- Examples are now tested in CI
- Some examples were not building

### CODE REVIEW

File: intro/http-client/examples/https_client.rs [hunk 1]
```diff
@@ -37,6 +37,7 @@ fn get(url: impl AsRef<str>) -> anyhow::Result<()> {
     // 1. create a new EspHttpClient with SSL certificates enabled
     let mut client = EspHttpClient::new(&EspHttpClientConfiguration {
         use_global_ca_store: true,
+        // TODO: Fix this
```
Comment (MabezDev) [inline intro/http-client/examples/https_client.rs:40]: What needs fixing here?

**Review by MabezDev — APPROVED** [review]
LGTM! Thanks for taking care of this :). One quick comment about a TODO but it's no big deal.

File: intro/http-client/examples/https_client.rs (hunk 1 above)
Comment (SergioGasquez) [reply intro/http-client/examples/https_client.rs:40]: For some reason, `esp_idf_sys::esp_crt_bundle_attach` (is on the next line) was not found. I could not investigate it too much tbh, will come back at it next week.

Comment (MabezDev) [reply intro/http-client/examples/https_client.rs:40]: Gotcha thanks!

**Review by MabezDev — APPROVED** [review]

File: intro/http-client/examples/https_client.rs (hunk 1 above)
Comment (SergioGasquez) [reply intro/http-client/examples/https_client.rs:40]: I've created https://github.com/esp-rs/espressif-trainings/issues/136 so I don't forget about it

**Pull request merged**